lgn-provers/src/params/mod.rs

```diff
@@ -1,330 +1,153 @@
-
-        let expected_hashes_file = Path::new(&checksum_expected_local_path);
-        let expected_hashes = read_hashes(
-            &mut std::io::stderr(),
-            &(
-                "output".to_string(),
-                expected_hashes_file.to_path_buf(),
-            ),
-        )
-        .unwrap_or_else(
-            |_| {
-                panic!(
-                    "while comparing hashes to `{}`",
-                    expected_hashes_file.display()
-                )
-            },
-        );
-
-        let expected_hash: BTreeMap<String, String> = expected_hashes
-            .iter()
-            .filter(|hash| hash.0 == file_name)
-            .map(
-                |hash| {
-                    (
-                        hash.0
-                            .to_owned(),
-                        hash.1
-                            .to_owned(),
-                    )
-                },
-            )
-            .collect();
-        let compare_hashes = compare_hashes(
-            "compare_hashes",
-            computed_hash.clone(),
-            expected_hash.clone(),
-        );
-
-        let result = write_hash_comparison_results(
-            &mut std::io::stdout(),
-            &mut std::io::stderr(),
-            compare_hashes.clone(),
-        );
-
-        debug!(
-            "Checksum. file: {:?} computed_hashes: {:?} computed_hash: {:?} expected_hashes: {:?} expected_hash: {:?} compare_hashes: {:?}",
-            file,
-            computed_hashes,
-            computed_hash,
-            expected_hashes,
-            expected_hash,
-            compare_hashes,
-        );
-
-        match result {
-            checksums::Error::NoError => {
-                info!(
-                    "Checksum is successful. file: {:?} computed_hash: {:?} expected_hash: {:?} compare_hashes: {:?}",
-                    file,
-                    computed_hash,
-                    expected_hash,
-                    compare_hashes,
-                );
-                Ok(true)
-            },
-
-            _ => {
-                warn!(
-                    "Checksum failed. file: {:?} computed_hash: {:?} expected_hash: {:?} compare_hashes: {:?}",
-                    file,
-                    computed_hash,
-                    expected_hash,
-                    compare_hashes,
-                );
-                if let Err(err) = std::fs::remove_file(Path::new(file)) {
-                    error!(
-                        "Error deleting file {:?}: {}",
-                        file, err
-                    );
-                }
-                Ok(false)
-            },
-        }
-    }
-
-    fn store_file(
-        file: &Path,
-        params: &Bytes,
-    ) -> anyhow::Result<()> {
-        info!(
-            "Storing params to local storage: {:?}",
-            file
+        let bytes = response.bytes().context("fetching params bytes")?;
+        let found_checksum = blake3::hash(&bytes);
```

Comment:
No big deal / blocker at all - but there are many ways to use blake3.
* I think you should enable the rayon feature ? It seems [it's disabled by default but enabled on doc.rs](https://docs.rs/blake3/1.5.5/blake3/struct.Hasher.html#method.update_rayon) - i think you can call `update_rayon` after, if it's not enabled by default in what you do, haven't checked.
* Have you considered streaming reqwest directly to a file and blake3 it with mmap ?

Reply:
> I think you should enable the rayon feature ?

Good point, I didn't.

> Have you considered streaming reqwest directly to a file and blake3 it with mmap ?

I'm not sure what would be the advantage of `web -> memory -> file -> mmap -> blake3` vs. `web -> memory -> blake3 -> file`?